vowpalwabbit/options_serializer_boost_po.cc

```diff
@@ -4,11 +4,19 @@
 
 #include "options_serializer_boost_po.h"
 
+#include <memory>
+
 using namespace VW::config;
 
 std::string options_serializer_boost_po::str() { return m_output_stream.str(); }
 
-const char* options_serializer_boost_po::data() { return m_output_stream.str().c_str(); }
+std::unique_ptr<const char> options_serializer_boost_po::data() {
+  auto str = m_output_stream.str();
+  char* buffer = new char[str.size() + 1];
+  str.copy(buffer, str.size());
+  buffer[str.size()] = '\0';
+  return std::unique_ptr<const char>(buffer);
```

Comment:
Since we are doing a copy here anyways, can we change the interface to be a `std::string`?

Reply:
Potentially, but I think the idea is to get a raw binary representation of the object? Although this function is actually not used anywhere.. Should we instead just remove it?

Reply:
If it is unused, I would remove it. Encapsulation, particularly around configuration is good - it could make it easier to pull boost::po into just the executable, rather than the lib.

Reply:
Great, I agree. Will remove.